be/src/vec/common/int_exp.h

```diff
@@ -22,6 +22,34 @@
 
 #include <cstdint>
 #include <limits>
+#include <utility>
+
+namespace exp_details {
+
+// compile-time exp(v, n) by linear recursion
+template <typename T, T v, std::size_t n>
+constexpr inline const T exp = v* exp<T, v, n - 1>;
```

Comment:
Seems here have a bug of clang-format, maybe we can use `T(v) * exp<T, v, n - 1>` to work around.

Reply:
done